src/diceBot/AnimaAnimus.rb

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# frozen_string_literal: true
+
+require 'utils/table.rb'
+require 'utils/range_table'
+require 'utils/ArithmeticEvaluator'
+
+class AnimaAnimus < DiceBot
+  # ゲームシステムの識別子
+  ID = 'AnimaAnimus'
+
+  # ゲームシステム名
+  NAME = 'アニマアニムス'
+
+  # ゲームシステム名の読みがな
+  #
+  # 「ゲームシステム名の読みがなの設定方法」（docs/dicebot_sort_key.md）を参考にして
+  # 設定してください
+  SORT_KEY = 'あにまあにむす'
+
+  # ダイスボットの使い方
+  HELP_MESSAGE = <<MESSAGETEXT
+・行為判定(xAN<=y±z)
+　十面ダイスをx個振って判定します。達成値が算出されます(クリティカル発生時は2増加)。
+　x：振るダイスの数。魂魄値や攻撃値。
+　y：成功値。
+　z：成功値への補正。省略可能。
+　(例) 2AN<=3+1 5AN<=7
+・各種表
+　情報収集表　IGT/喪失表　LT
+MESSAGETEXT
+
+  def initialize
+    super
+    @sendMode = 2
+  end
+
+  def rollDiceCommand(command)
+    case command
+    when /(\d+)AN<=(\d+([\+\-]\d+)*)/i
+      return check_action(Regexp.last_match)
+    else
+      return roll_tables(command, TABLES)
+    end
+  end
+
+  def check_action(match_data)
+    a = ArithmeticEvaluator.new()
+    dice_cnt = a.eval(match_data.values_at(1)[0])
+    target = a.eval(match_data.values_at(2)[0])
+    debug("dice_cnt", dice_cnt)
+    debug("target", target)
+
+    _, dice_str, = roll(dice_cnt, 10)
+    dice_arr = dice_str.split(',')
+    suc_cnt = 0
+    has_critical = false
+
+    i = 0
+    while i < dice_arr.length
+      if dice_arr[i].to_i <= target
+        suc_cnt += 1
+      end
+      if dice_arr[i].to_i == 1
+        has_critical = true
+      end
+      i += 1
+    end
```

Comment:
```suggestion
    _, dice_str, = roll(dice_cnt, 10)
    dice_arr = dice_str.split(',').map(&:to_i)
    suc_cnt = dice_arr.count { |x| x <= target }
    has_critical = dice_arr.include?(1)
```

Rubyではwhileやforを使ったループは行わずに、Array#each, Array#map, Array#count, Array#include? などを使うのが一般的です